internal/manager/run.go

```diff
@@ -184,23 +184,35 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic, d
 		// In case of failures when building Konnect related objects, we're not returning errors as Konnect is not
 		// considered critical feature, and it should not break the basic functionality of the controller.
 
+		konnectAdminAPIClient, err := adminapi.NewKongClientForKonnectRuntimeGroup(ctx, c.Konnect)
+		if err != nil {
+			setupLog.Error(err, "failed creating Konnect Runtime Group Admin API client, skipping synchronisation")
+		} else {
+			setupLog.Info("Initialized Konnect Admin API client")
+			clientsManager.SetKonnectClient(konnectAdminAPIClient)
+		}
+
 		setupLog.Info("Start Konnect client to register runtime instances to Konnect")
 		konnectNodeAPIClient, err := konnect.NewNodeAPIClient(c.Konnect)
 		if err != nil {
 			setupLog.Error(err, "failed creating konnect client, skipping running NodeAgent")
 		} else {
 			hostname, _ := os.Hostname()
 			version := metadata.Release
-			agent := konnect.NewNodeAgent(hostname, version, setupLog, konnectNodeAPIClient)
-			agent.Run()
-		}
+			// set channel to send config status.
+			configStatusChan := make(chan dataplane.ConfigStatus, 1)
+			configStatusNotifier := dataplane.NewChannelConfigNotifier(configStatusChan)
```

Comment:
If no one is going to use the channel outside of the notifier why don't we put its initialization to the "constructor" itself. This would simplify the logic at the call site (e.g. here). WDYT?

Reply:
moved to "constructor".